gravatar/src/main/java/com/gravatar/services/ServiceUtils.kt

```diff
@@ -10,12 +11,12 @@ internal inline fun <T> runCatchingRequest(block: () -> T?): Result<T, ErrorType
         if (result != null) {
             Result.Success(result)
         } else {
-            Result.Failure(ErrorType.NOT_FOUND)
+            Result.Failure(ErrorType.NotFound)
         }
     } catch (cancellationException: CancellationException) {
         throw cancellationException
     } catch (ex: Exception) {
-        Result.Failure(ex.errorType())
+        Result.Failure(ex.errorType(GravatarSdkContainer.instance.gson))
```

Comment:
❓ Why pass `gson` instance through all methods instead of getting it in `errorTypeFromHttpCode` where it's used?

Reply:
Generally, I didn't want to access a static Gson instance from an extension function that we have tests for. It wouldn't be a problem now, but if we needed to configure it differently for the tests we would have to rely on the `GravatarSdkContainerRule`